src/entity/posts/utils.ts

```diff
@@ -464,22 +464,32 @@ export const relatePosts = async ({
   return posts;
 };
 
-export const normalizeCollectionPostSources = async ({
+export const getDistinctSourcesBaseQuery = ({
   con,
   postId,
 }: {
   con: DataSource | EntityManager;
   postId: CollectionPost['id'];
-}) => {
-  const distinctSources = await con
+}) =>
+  con
     .createQueryBuilder()
-    .select('s.id as id')
     .from(PostRelation, 'pr')
     .leftJoin(Post, 'p', 'p.id = pr."relatedPostId"')
     .leftJoin(Source, 's', 's.id = p."sourceId"')
-    .where('pr."postId" = :postId', { postId: postId })
+    .where('pr."postId" = :postId', { postId })
```

Comment:
we also need:
```js
.andWhere('pr."type" = :type', { type: relationType })
```

and add `relationType: PostRelationType`. This will allow us to support other relations down the line.

Reply:
See 84e05aa

Reply:
👌